aip/general/0213/aip.md

```diff
@@ -0,0 +1,214 @@
+  objects to and from JSON.
+- [`google.protobuf.Struct`][struct]: JSON-like structures (a dictionary of
+  primitives, lists, and other dictionaries). The protobuf runtime provides
+  helper functions in most languages to convert `Struct` objects to and from
+  JSON.
+
+`google.protobuf.Struct` and `google.protobuf.Value` are designated common
+components by this AIP; proto-based APIs **should** use them when appropriate.
+
+<!-- prettier-ignore-start -->
+[datetime]: https://docs.python.org/3/library/datetime.html#datetime.datetime
+[duration]: https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/duration.proto
+[protobuf]: https://github.com/protocolbuffers/protobuf/tree/main/src/google/protobuf
+[struct]: https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/struct.proto
+[timedelta]: https://docs.python.org/3/library/datetime.html#datetime.timedelta
+[timestamp]: https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/timestamp.proto
+<!-- prettier-ignore-end -->
+
+## Libraries for common types
+
+For the common components in the `aip.type` namespace, which represent common
+data types, the [AIP common components][] repo may contain canonical libraries
+in a number of languages. These libraries are designed to be idiomatic in a
+given language, and should feel similar to using the language's standard
+libraries. They should provide basic functionality like adding two `Money`
+values, or determining if one `Date` comes before another.
+
+When a language already has a standard library representation of a common type
+(such as Python's `datetime` for the `Timestamp` type), there may instead be a
+library for converting the JSON or protobuf representation to the standard
+library representation.
+
+If a library you want does not exist, and you want to contribute one, please
+[open an issue][] on the [AIP common components][] repository in GitHub.
+
+## Appendix: Adding to common components
+
+Occasionally, it may be useful to add protos to these packages or to add to the
+list of common components. In order to do this, [open an issue][] on the [AIP
+common components][] repository in GitHub.
+
+However, some general guidelines are worth noting for this:
+
+- Schemas **should** only be granted common component status if we are certain
+  that they will never change (at all -- even in ways that would normally be
+  considered backwards compatible). Common components are generally not
+  versioned, and it must be the case that we can rely on the component to be a
+  complete and accurate representation indefinitely.
+- Schemas must be applicable to a significant number of APIs for consideration
+  as common components.
+- Even after a common component is added, APIs using local versions **must**
+  continue to do so until they go to the next major version.
+
+In the event that you believe adding a common component is appropriate, please
```

Comment:
redundant with the "open an issue" in the top of this section.

Reply:
Removed